apps/meteor/ee/app/livechat-enterprise/server/services/omnichannel.internalService.ts

```diff
@@ -40,7 +41,11 @@ export class OmnichannelEE extends ServiceClassInternal implements IOmnichannelE
 		if (room.onHold) {
 			throw new Error('error-room-is-already-on-hold');
 		}
-		if (room.lastMessage?.token) {
+		const restrictedOnHold = settings.get('Livechat_allow_manual_on_hold_upon_agent_engagement_only');
+		const canRoomBePlacedOnHold = !room.onHold;
+		const canAgentPlaceOnHold = !room.lastMessage?.token;
+		const canPlaceChatOnHold = canRoomBePlacedOnHold && (!restrictedOnHold || canAgentPlaceOnHold);
+		if (!canPlaceChatOnHold) {
 			throw new Error('error-contact-sent-last-message-so-cannot-place-on-hold');
```

Comment:
```suggestion
			throw new Error('error-cannot-place-chat-on-hold');
```

Reply:
I dont see why wee need a more generic error. The only possible error is error-contact-sent-last-message-so-cannot-place-chat-on-hold. All of the remaining settings scenarios will allow the agent to place the chat on hold.